client/gribiclient.go

```diff
@@ -345,7 +347,23 @@ func (c *Client) Connect(ctx context.Context) error {
 				log.V(2).Infof("shutting down recv goroutine")
 				return
 			}
-			if done := respHandler(stream.Recv()); done {
+			resp, err := stream.Recv()
+			trailer := stream.Trailer()
+
+			if len(trailer) != 0 {
+				if url, ok := trailer["errordetails-typeurl"]; ok {
+					anyObj := &anypb.Any{TypeUrl: url[0], Value: []byte(trailer.Get("errordetails-value-bin")[0])}
```

Comment:
If `trailer.Get` doesn't return anything here, I think this will panic. Can you check that both the URL and the binary data are actually returned?

Reply:
Taken care in latest commit